synapse/storage/databases/main/experimental_features.py

```diff
@@ -34,7 +33,7 @@ def __init__(
         super().__init__(database, db_conn, hs)
 
     @cached()
-    async def list_enabled_features(self, user_id: str) -> StrCollection:
+    async def list_enabled_features(self, user_id: str) -> FrozenSet[str]:
```

Comment:
Hmm. Should we make the plugin consider StrCollection to be immutable?

Reply:
We talked about this in DM, but I don't think so.